### ISSUE TITLE
Complete level 25

### ISSUE THREAD

**Issue by j0n3lson** [j0n3lson/codenext-clubs-linux-admin#2]
https://overthewire.org/wargames/bandit/bandit26.html

**Comment by j0n3lson** [comment]
### I'm not using the `/bin/bash` or `/bin/sh` shells when logging into level 26.

The shell a user is mapped to is given in `/etc/passwd`. In this case, it's a shell script:
```shell
# In this case it's not your usual shell
bandit25@bandit:~$ grep "bandit26" /etc/passwd
bandit26:x:11026:11026:bandit level 26:/home/bandit26:/usr/bin/showtext
bandit25@bandit:~$ 

# Which is a shell script.
bandit25@bandit:~$ grep "bandit26" /etc/passwd
bandit26:x:11026:11026:bandit level 26:/home/bandit26:/usr/bin/showtext
bandit25@bandit:~$ 

# And where are the contents
bandit25@bandit:~$ cat /usr/bin/showtext
#!/bin/sh

export TERM=linux

more ~/text.txt
exit 0
bandit25@bandit:~$ 
```
I actually found `/usr/bin/showtext` on accident as I was looking for the *alternative* shells the level instructions hinted it. My thinking was that maybe if I found the binary I could exploit something with the binary (maybe permissions, maybe I could replace it?)

**Comment by j0n3lson** [comment]
Okay a little more progress. Since I know the shell script traps me in the `more` command, I started looking at `more` itself. It turns out that you can specify additional options that `more` uses bye setting the `MORE` environment variable from `man more`:

```
OPTIONS
       Options are also taken from the environment variable MORE (make sure to precede them with a dash (-)) but command-line options will override those.
```

So I started looking at how to set/send environment variables for an `ssh` session. This lead me to the `ssh -o` option which allows sending various options. One of those options is `SendEnv` which lets you send the value of a given environment variable.

I was very confused about how to use `SendEnv` through trial and error I got the syntax right:

> ssh -o SendVar=<VAR_NAME> ...

For example:
```shell
export MORE="-1"
ssh -o SendEnd=MORE ...
```
If you turn on verbose `ssh` output with the `-vvvv` flag you can see the variable being sent:
```shell
...
debug3: Ignored env OLDPWD
debug3: Ignored env USER
debug1: Sending env MORE = -1
...
```
### Testing an escape

So I was curious about what would happen if I *could* change the value of the `MORE` variable. How would it change the behavior of more. So I did a test:

![escape_more](https://user-images.githubusercontent.com/620536/69503810-942b2d00-0ed2-11ea-981f-e90df3b0252b.gif)

Okay, that's great. If I can only get `more` to go into interactive mode then I can run any command I want including `/bin/sh` in order to get a legit shell.

But how do I get that interactive shell!?

**Comment by j0n3lson** [comment]
Turns out I was on the right track. I just needed to resize the window so it was less than the line count. This puts `more` in interactive mode. From there, I can open vi by hitting `v` on the keyboard and the go into edit mode to open the password file!

**Comment by j0n3lson** [comment]
Fixed in f92637ecc5e91624b6527ce2aa953a3b943540d3
